lib/virtio_mmio/virtio_mmio_drv.c

```diff
@@ -325,7 +325,7 @@ struct virtqueue *virtio_mmio_setup_virtqueue(struct virtio_device *vdev,
 	virtio_mmio_write32(vdev, VIRTIO_MMIO_QUEUE_NUM, vq->vq_nentries);
 	virtio_mmio_write32(vdev, VIRTIO_MMIO_QUEUE_ALIGN, 4096);
 	virtio_mmio_write32(vdev, VIRTIO_MMIO_QUEUE_PFN,
-			    ((uintptr_t)metal_io_virt_to_phys(vq->shm_io,
+			    ((uintptr_t)metal_io_virt_to_phys(vmdev->shm_io,
```

Comment:
The two source snippets are logically identical because vq->shm_io is set to vmdev->shm_io in the call to virtqueue_set_shmem_io a few line above.

Reply:
Yes they are, this is not meant to be a functional change. The point is to not need to poke into the `vq` struct as part of a larger restructuring. Also the value `shm_io` inside `vq` is meant for the buffers, and here they use it for the vring pointers themselves, which only works in some cases (when they are both part of one large IO region), it is more correct to not assume this will always be true.

Reply:
@danmilea 
any concern or it is ok for you?

Reply:
This commit doesn't change functionality in any way. It's OK to merge from my point of view.